pkg/combustion/elemental.go

```diff
@@ -0,0 +1,88 @@
+package combustion
```

Comment:
Shouldn't this check if the rpms are in place as well? (maybe it is and I couldn't find it)

Reply:
You're correct. We can't fail during build time if the components are missing and we need the validation.

Reply:
Totally agree we should check. Problem is, it's not that simple. Here are the options I see:

= Option #1: Force the user to name the files according to our convention, so something like `elemental-toolkit.rpm` =
Pros:
- Makes validation simple
- Continues to use the existing RPM install functionality at no extra cost to us
Cons:
- Adds another manual step for the user to do
- Removes version/arch/whatever else information from the RPM name
- Relies on documentation to convey the convention, as compared to Option #3 which is more explicit

= Option #2: Have the validation logic only check for the prefix of the RPM, making sure there are RPMs that start with `elemental-toolkit` and `elemental-system-agent`
Pros:
- Continues to use the existing RPM install functionality at no extra cost to us
- User shouldn't need to do any extra renaming steps in most cases
Cons:
- Validation is a bit on the fragile side
- Relies on documentation to convey the convention, as compared to Option #3 which is more explicit

= Option #3: Have the user specify the names of these files in the image definition, and still adding =
Pros:
- Can still use the existing RPM functionality; the image just won't be built if EIB can't find them by name
Cons:
- Extra data needs to be added to the image definition. _(jdob: I don't hate that; the magic of just dropping it in `rpms` is nice but there is something to be said about having the elemental pieces not work on magic)_

= Option #4: Have the user create an `elemental` directory in the image config dir and include the RPMs in there
Pros:
- Arguably clean, since it makes it obvious what those RPMs are in there for as compared to the catch-all `rpms` directory
- Doesn't use the existing RPM functionality at all, instead injecting the names of the RPMs into the template for elemental registration, thus removing the cross-component dependency
Cons:
- Still need to figure out how to resolve the potential name differences (see Options 1 and 2)
- Potentially confusing why there would be RPMs outside of the `rpms` directory _(jdob: I don't think this is a huge issue)_

Thoughts?

Reply:
Thanks for the comprehensive and thoughtful considerations! I really prefer option 5 but it also implies that we know exactly where to grab the RPMs from which has to be further investigated.

Approving for now and we can introduce the validation in a follow up.